### PR TITLE
Simplify `_extract_and_set_api_key`

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Simplify `_extract_and_set_api_key`.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/gateway/config.py [hunk 1]
```diff
@@ -114,21 +114,18 @@ def _extract_and_set_api_key(config, provider):
...
-                config_dict[key] = _resolve_api_key_from_input(config_dict[key])
+                value = _resolve_api_key_from_input(value)
+                return config_class(**{**config.dict(), key: value})
 
-    return config_types[provider](**config_dict)
+            raise MlflowException.invalid_parameter_value(
+                f"For the {provider} provider, the api key must either be specified within the "
+                "configuration supplied or an environment variable set whose key is "
+                "defined within the configuration"
+            )
```
Comment (harupy) [inline mlflow/gateway/config.py:128]: This sounds like the api key is not optional, but in our config data models, it is.

File: mlflow/gateway/config.py [hunk 2]
```diff
@@ -114,21 +114,18 @@ def _extract_and_set_api_key(config, provider):
...
 
-    config_dict = config.dict()
-
     for config_class, key in required_keys.items():
         if isinstance(config, config_class):
-            if getattr(config, key, None) is None:
-                raise MlflowException.invalid_parameter_value(
-                    f"For the {provider} provider, the api key must either be specified within the "
-                    "configuration supplied or an environment variable set whose key is "
-                    "defined within the configuration"
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/gateway/config.py:125]: I'd say "the api key is missing" here, then throw a more detailed error message in `_resolve_api_key_from_input`.

[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/gateway/config.py:125]: I won't fix this in this PR.

File: mlflow/gateway/config.py (hunk 1 above)
Comment (BenWilson2) [reply mlflow/gateway/config.py:128]: Good point. I'm not sure if we're going to be exposing unsecured endpoints through the gateway service (at least not for this MVP). We can always add conditional logic later and push the validation to a per-provider level for presence enforcement outside of Pydantic later on if we need to.

**Review by BenWilson2 — APPROVED** [review]
LGTM

**Pull request merged**